universal-application-tool-0.0.1/app/services/applicant/Block.java

```diff
@@ -0,0 +1,11 @@
+package services.applicant;
+
+import com.google.auto.value.AutoValue;
+
+/** Represents a block in the context of a specific user's application. */
+@AutoValue
+public abstract class Block {
+
+  /** The block's ID. Note this is different from the BlockDefinition's ID. */
+  abstract long id();
```

Comment:
What does block id represent?

Reply:
It seems like repeated fields will require `Block`s to have their own set of IDs, distinct from those of `BlockDefinition`. 😕 

`id` will help us map URLs to the right block and vice versa. We haven't thought through exactly how/when we'll generate these `id`s yet though.

Reply:
suggestion (non-blocking): Consider expanding a bit on why these `id`s are different in the docstring.

Reply:
done